fv/test-workload/test-workload.go

```diff
@@ -394,60 +393,34 @@ func main() {
 		for _, port := range ports {
 			var myAddr string
 			if strings.Contains(ipAddress, ":") {
-				myAddr = "[" + ipAddress + "]:" + port
+				myAddr = "[" + ipAddress + "]"
 			} else {
-				myAddr = ipAddress + ":" + port
+				myAddr = ipAddress
+			}
+			if !strings.HasPrefix(protocol, "ip") {
+				myAddr += ":" + port
 			}
 			logCxt := log.WithFields(log.Fields{
-				"udp":    udp,
-				"sctp":   useSctp,
-				"myAddr": myAddr,
+				"protocol": protocol,
+				"myAddr":   myAddr,
 			})
-			if udp {
+			if strings.HasPrefix(protocol, "ip") {
+				logCxt.Info("About to listen for raw IP packets")
+				p, err := net.ListenPacket(protocol, myAddr)
+				panicIfError(err)
+				logCxt.Info("Listening for raw IP packets")
+
+				go loopRespondingToPackets(logCxt, p)
+			} else if protocol == "udp" {
 				// Since UDP is connectionless, we can't use Listen() as we do for TCP.  Instead,
 				// we use ListenPacket so that we can directly send/receive individual packets.
 				logCxt.Info("About to listen for UDP packets")
 				p, err := net.ListenPacket("udp", myAddr)
 				panicIfError(err)
 				logCxt.Info("Listening for UDP connections")
 
-				go func() {
```

Comment:
This code just moved down below as a named function (so it can be used for UDP and raw IP)